concepts/objects/about.md

```diff
@@ -0,0 +1,260 @@
+
+Further, the _name_ must be a string.
+Another word for _name_ is _key_.
+
+The _value_ can be of any JSON type.
+Different keys in the object can have values of different type, like this example.
+
+```json
+{
+  "name": "Jane",
+  "age": 42,
+  "pets": ["cat", "fish"],
+  "address": {"street": "123 Main St", "city": "Springfield"}
+}
+```
+
+Note that there **must not** be a comma following the _last_ key-value pair.
+
+## Creating objects
+
+Use braces to collect the name/value pairs.
+Even though the names must be strings, they do not need to be quoted if the names are "identifier-like" (composed of alphanumeric characters and underscore, and not started with a digit).
+
+```jq
+{name: "Jane", age: 42}
+```
+
+It is valid to use keys that are not "identifier-like".
+Just quote them.
+
+```jq
+{"first name": "Jane", "last name": "Lopez", age: 42}
+```
+
+If the name is the result of an expression, the expression **must** be in parentheses.
+
+```sh
+$ echo "Jane" | jq -Rc '{.: 42}'
+jq: error: syntax error, unexpected '.' (Unix shell quoting issues?) at <top-level>, line 1:
+{.: 42}
+jq: error: May need parentheses around object key expression at <top-level>, line 1:
+{.: 42}
+jq: 2 compile errors
+
+$ echo "Jane" | jq -Rc '{(.): 42}'
+{"Jane":42}
+```
+
+### Short-hand notations
+
+A couple of frequent cases for object construction have short-hand syntax.
+
+1. Re-using key-value pairs from an input object.
+
+   ```sh
+   $ jq -n '$ENV | {USER: .USER, EDITOR: .EDITOR}'
+   {
+     "USER": "glennj",
+     "EDITOR": "vim"
+   }
+   ```
+
+   `$ENV` is a object holding the name-value pairs of the environment.
+
+   The repetitive `key: .key` pairs can be written as just `key`
+
+   ```sh
+   $ jq -n '$ENV | {USER, EDITOR}'
+   {
+     "USER": "glennj",
+     "EDITOR": "vim"
+   }
+   ```
+		
+1. Similarly, there's a short-hand when using variables.
+
+   ```sh
+   $ jq -cn '"foo" as $a | "bar" as $b | {a: $a, b: $b}'
+   {"a":"foo","b":"bar"}
+
+   $ jq -cn '"foo" as $a | "bar" as $b | {$a, $b}'
+   {"a":"foo","b":"bar"}
+   ```
+
+## Indexing
+
+Values are retrieved from an object with _dot notation_.
+
+```jq
+{name: "Jane", age: 42} | .age    # => 42
+```
+
+If you cannot refer to the key as an identifier, use _bracket notation_.
+
+```jq
+"name" as $key | {name: "Jane", age: 42} | .$key    # => error
+"name" as $key | {name: "Jane", age: 42} | .[$key]  # => "Jane"
+```
+
+If the key is not in the object, the result is `null`
+
+```jq
+{name: "Jane", age: 42} | .sport  # => null
+```
+
+### Indexing nested values
+
+You will discover that this is an action you need to do again and again when parsing results from API queries.
+
+Given an object that contains array or object values, we can "chain" index expressions together.
+
+```jq
+{
+  "name": "Jane",
+  "age": 42,
+  "pets": ["cat", "fish"],
+  "address": {"street": "123 Main St", "city": "Springfield"}
+} as $example
+|
```

Comment:
Would this example be simpler without using variables?

Reply:
Do you mean something like this
```sh
$ cat example.json
{
  "name": "Jane",
  "age": 42,
  "pets": ["cat", "fish"],
  "address": {"street": "123 Main St", "city": "Springfield"}
}
$ jq '.pets[1]' example.json        # => "fish"
$ jq '.address.city' example.json   # => "Springfield"
```

Reply:
```
{
  "name": "Jane",
  "age": 42,
  "pets": ["cat", "fish"],
  "address": {"street": "123 Main St", "city": "Springfield"}
}
| (
  .pets[1],
  .addresss.city
)
```